src/components/app/details/appDetails/GraphsModal.tsx

```diff
@@ -86,15 +88,29 @@ export class GraphModal extends Component<GraphModalProps, GraphModalState>{
     }
 
     getNewGraphs(tab: AppMetricsTabType) {
-        let cpu = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'cpu', this.props.newPodHash, this.state.calendarInputs, tab, false);
-        let ram = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'ram', this.props.newPodHash, this.state.calendarInputs, tab, false);
-        let latency = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'latency', this.props.newPodHash, this.state.calendarInputs, tab, false);
-        let status2xx = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'status', this.props.newPodHash, this.state.calendarInputs, tab, false, '2xx');
-        let status4xx = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'status', this.props.newPodHash, this.state.calendarInputs, tab, false, '4xx');
-        let status5xx = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'status', this.props.newPodHash, this.state.calendarInputs, tab, false, '5xx');
-        let status = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'status', this.props.newPodHash, this.state.calendarInputs, tab, false, 'Throughput');
-        let throughput = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, 'status', this.props.newPodHash, this.state.calendarInputs, tab, false, 'Throughput');
-        let mainChartUrl = getIframeSrc(this.props.appId, this.props.envId, this.props.environmentName, this.state.mainChartName, this.props.newPodHash, this.state.calendarInputs, tab, true, this.state.statusCode);
+        let k8sVersion = this.props.k8sVersion;
+        let defaultK8sVersion = 'v16.1.10';
+        if (!isK8sVersionValid(k8sVersion)) {
+            k8sVersion = defaultK8sVersion;
+        }
+        let appInfo = {
+            appId: this.props.appId,
+            envId: this.props.envId,
+            environmentName: this.props.environmentName,
+            newPodHash: this.props.newPodHash,
+            k8sVersion: k8sVersion,
+        }
+
+        let cpu = getIframeSrc(appInfo, 'cpu', this.state.calendarInputs, tab, false);
```

Comment:
generally we should extract out constants like 'cpu', 'ram', 'latency' at the top of the file